Algorithm.Python/ObjectStoreExampleAlgorithm.py

```diff
@@ -0,0 +1,102 @@
+from datetime import timedelta
+from io import StringIO
+import pandas as pd
+
+class ObjectStoreExampleAlgorithm(QCAlgorithm):
+    '''This algorithm showcases some features of the IObjectStore feature.
+    One use case is to make consecutive backtests run faster by caching the results of
+    potentially time consuming operations. In this example, we save the results of a
+    history call. This pattern can be equally applied to a machine learning model being
+    trained and then saving the model weights in the object store.
+    '''
+    SPY_Close_ObjectStore_Key = "spy_close"
+    SPY_Close_History = RollingWindow[IndicatorDataPoint](252)
+    SPY_Close_EMA10_History = RollingWindow[IndicatorDataPoint](252)
+    SPY_Close_EMA50_History = RollingWindow[IndicatorDataPoint](252)
+
+    def Initialize(self):
+        self.SetStartDate(2013, 10, 7)
+        self.SetEndDate(2013, 10, 11)
+
+        self.SPY = self.AddEquity("SPY", Resolution.Minute).Symbol
+
+        self.SPY_Close = self.Identity(self.SPY, Resolution.Daily)
+        self.SPY_Close_EMA10 = IndicatorExtensions.EMA(self.SPY_Close, 10)
+        self.SPY_Close_EMA50 = IndicatorExtensions.EMA(self.SPY_Close, 50)
+
+        # track last year of close and EMA10/EMA50
+        self.SPY_Close.Updated += lambda _, args: self.SPY_Close_History.Add(args)
+        self.SPY_Close_EMA10.Updated += lambda _, args: self.SPY_Close_EMA10_History.Add(args)
+        self.SPY_Close_EMA50.Updated += lambda _, args: self.SPY_Close_EMA50_History.Add(args)
+
+        if self.ObjectStore.ContainsKey(self.SPY_Close_ObjectStore_Key):
+            # our object store has our historical data saved, read the data
+            # and push it through the indicators to warm everything up
+            values = self.ObjectStore.Read(self.SPY_Close_ObjectStore_Key)
+            self.Debug(f'{self.SPY_Close_ObjectStore_Key} key exists in object store.')
+
+            history = pd.read_csv(StringIO(values), header=None, index_col=0, squeeze=True)
+            history.index = pd.to_datetime(history.index)
+            for time, close in history.iteritems():
+                self.SPY_Close.Update(time, close)
+
+        else:
+            self.Debug(f'{self.SPY_Close_ObjectStore_Key} key does not exist in object store. Fetching history...')
+
+            # if our object store doesn't have our data, fetch the history to initialize
+            # we're pulling the last year's worth of SPY daily trade bars to fee into our indicators
+            history = self.History(self.SPY, timedelta(365), Resolution.Daily).close.unstack(0).squeeze()
+
+            for time, close in history.iteritems():
+                self.SPY_Close.Update(time, close)
+
+            # save our warm up data so next time we don't need to issue the history request
+            self.ObjectStore.Save(self.SPY_Close_ObjectStore_Key,
+                '\n'.join([f'{x.EndTime},{x.Value}' for x in sorted(self.SPY_Close_History, key=lambda x: x.EndTime)]))
```

Comment:
A couple of doubts:
- Interesting that we have to sort the RollingWindow, shouldn't it already be sorted? Seems we are doing the same for C# though
- Wondering if we could directly store the RollingWindow data points, not creating an intermediate string item?

Reply:
- The `RollingWindow` objects are sorted in reverse order. The most recent item is found in index 0.
- We may store it directly, but how would we read the storage? Thinking about that, I choose `csv` format because pandas can read it. We need to come up with an example using a machine learning model.

Reply:
- Rolling window: maybe we should use `reverse` for C# and Py instead, else its kind of implying the data isn't sorted at all
- Seems we need a specific issue (or we could use this one) to add python Tests for this and fix/add new overloads if required